### PR TITLE
Clarify the effects of path separator

### PR DESCRIPTION
**Pull request by xuhdev**
This is consistent with the rules from `.gitignore`: <https://git-scm.com/docs/gitignore>

> If there is a separator at the beginning or middle (or both) of the
> pattern, then the pattern is relative to the directory level of the
> particular .gitignore file itself. Otherwise the pattern may also match
> at any level below the .gitignore level.

Since we intend to be consistent with `.gitignore`, this part of the spec has been unclear.

Fix editorconfig/editorconfig#509

----
📚 Documentation preview 📚: https://editorconfig-specification--49.org.readthedocs.build/

### CODE REVIEW

**Comment by xuhdev** [comment]
And we have been unclear with slashes at the end because it didn't matter as matching directories is pointless...

File: index.rst [hunk 1]
```diff
@@ -160,6 +160,13 @@ special characters for wildcard matching:
      - any integer numbers between ``num1`` and ``num2``, where ``num1`` and ``num2``
        can be either positive or negative
 
+If there is a path separator (``/``) at in the glob, then the glob is relative
```
Comment (cxw42) [inline index.rst:163]: Suggestions:
```suggestion
If the glob contains a path separator (a ``/`` not inside square brackets), then the glob is relative
```

File: index.rst [hunk 2]
```diff
@@ -160,6 +160,13 @@ special characters for wildcard matching:
...
+to the directory level of the particular `.editorconfig` file itself.
+Otherwise the pattern may also match at any level below the `.editorconfig`
+level. For example, ``*.c`` matches any file that ends with ``.c`` in the
+directory of ``.editorconfig``, but ``subdir/*.c`` only matches files that end
+with ``.c`` in the ``subdir`` directory in the directory of ``.editorconfig``.
+
```
Comment (cxw42) [inline index.rst:169]: @xuhdev You mentioned

> we have been unclear with slashes at the end

That is mentioned in https://github.com/editorconfig/editorconfig/issues/493 .  Would you be willing to fix that as part of this PR as well?  https://github.com/editorconfig/editorconfig/issues/493#issue-1977582345 has some proposed text.  (If you would rather it be in a separate PR, that's fine.)

**Review by cxw42 — APPROVED** [review]
Looks good!  A few comments.

Depending on whether the tests break existing behaviour, we might need to bump the minor version.

File: index.rst (hunk 2 above)
Comment (xuhdev) [reply index.rst:169]: (Let's do it in a separate PR)
